### ISSUE TITLE
[Review]: Improved Multi-label Classification under Temporal Concept Drift: Rethinking Group-Robust Algorithms in a Label-Wise Setting

### ISSUE THREAD

**Issue by dennislx** [dennislx/EvoNLP#3]
### Description

Chalkidis, I., & Søgaard, A. (2022). Improved Multi-label Classification under Temporal Concept Drift: Rethinking Group-Robust Algorithms in a Label-Wise Setting. arXiv preprint arXiv:2203.07856.

- say we want to associate medical records with diagnostic and procedure labels (Johnson et al., 2017), label scientific publications with ontological concepts (Tsatsaronis et al., 2015), legislation with relevant legal concepts (Mencia and Fürnkranzand, 2007) or categorize product descriptions (Lewis et al., 2004).
- there are two challenges in multi-label classification: 1) class imbalance and 2) class distribution changes over time. 

## Temporal Drift Research

1. Dumais, S., & Chen, H. (2000, July). [Hierarchical classification of web content](https://dl.acm.org/doi/abs/10.1145/345508.345593). In Proceedings of the 23rd annual international ACM SIGIR conference on Research and development in information retrieval (pp. 256-263).

2. Huang, X., & Paul, M. (2019, July). [Neural temporality adaptation for document classification: Diachronic word embeddings and domain adaptation models](https://aclanthology.org/P19-1403/). In Proceedings of the 57th Annual Meeting of the Association for Computational Linguistics (pp. 4113-4123).

3. Lukes, J., & Søgaard, A. (2018, October). [Sentiment analysis under temporal shift](https://aclanthology.org/W18-6210/). In Proceedings of the 9th workshop on computational approaches to subjectivity, sentiment and social media analysis (pp. 65-71).

4. Lazaridou, A., Kuncoro, A., Gribovskaya, E., Agrawal, D., Liska, A., Terzi, T., ... & Blunsom, P. (2021). [Pitfalls of static language modelling](https://arxiv.org/abs/2102.01951). arXiv preprint arXiv:2102.01951.

## Datasets

1. [United Kingdom (UK) legislation](https://zenodo.org/record/6355465/) where most of the laws have been categorized in thematic categories (e.g., health-care, finance, education, transportation, planing) that are presented in the document preamble. The dataset is chronologically split in training (20k, 1975–2002), development (8.5k, 2002–2008), test (8.5k, 2008–2018)
subsets. Version 1.0 has 18 labels (see uk-lex18.jsonl) and Version 2.0 has 69 labels (see uk-lex69.jsonl)

2. [European Union (EU) legislation](https://zenodo.org/record/5363165/). The dataset is chronologically split in training (55k, 1958–2010), development(5k, 2010–2012), test (5k, 2012–2016) subsets. We use the 1st and 2nd level of the EuroVoc taxonomy including 21 and 127 categories, respectively.

3. [The BIOASQ, task a:  Large-Scale Online Biomedical Semantic Indexing](http://participants-area.bioasq.org/datasets/) consists of biomedical articles from PubMed, annotated with concepts from the Medical Subject Headings (MeSH) taxonomy including 29k concepts. We subsample documents in the period 2000-2021, and we consider chronologically split training (80k, 1964–2015), development (10k, 2015–2018), test (10k, 2018–2020) subsets. We use the 1st and 2nd levels of MeSH, including 16 and 116 categories

### Contributions

- Previous works haven't considered class imbalance combined with temporal concept drift. 
- This paper focus on large-scale, multi-label document classification
- This paper reframes group-robust optimization algorithms as adaptation algorithms under concept drift. see their [open-sourced work](https://github.com/coastalcph/lw-robust)

Sagawa, S., Koh, P. W., Hashimoto, T. B., & Liang, P. (2019). [Distributionally robust neural networks for group shifts: On the importance of regularization for worst-case generalization](https://arxiv.org/abs/1911.08731). arXiv preprint arXiv:1911.08731.

## Fine-tuning Algorithms
- loss: binary cross entropy among $L$ classes for each instance (label A vs label non-A). 
$$\mathcal{L}(x) = -y \log \hat y - (1-y) \log(1-\hat y)$$
- loss: empirical risk minimization is the average bce (G is the number of labels)
$$\mathcal{L}(ERM) = \frac{1}{N}\sum_{j=1}^{N}\mathcal{L}(x_j)$$
- loss: group-wise losses, an equal number of instances are sampled per target class for each comparing algorithms
$$\mathcal{L}(g_i) = \frac{1}{N_{g_i}}\sum_{j=1}^{N_{g_i}}\mathcal{L}(x_j)$$
- group uniform loss: all labels are considered equally important
$$\mathcal{L}(GM) = \frac{1}{G}\sum_{j=1}^{G}\mathcal{L}(g_j)$$
- group distributionally robust optimization: labels with worse performance are weighted more
$$\mathcal{L}(DRO) = \frac{1}{G}\sum_{j=1}^{G}w_{g_j}\mathcal{L}(g_j)$$
- risk extrapolation: variation is more robust than expectation, so please consider variance across the label-wise losses
$$\mathcal{L}(REX) = \mathcal{L}(ERM) + \lambda \cdot \text{Var}\left(\mathcal{L}\_{g_1}, \mathcal{L}\_{g_2}, \cdots, \mathcal{L}\_{g_G}\right)$$
- invariant risk minimization: introduce group penalty terms that encourage performance of samples belonging to one label is similar (low variance)
- deep coral: minimize difference in second-order statistics (covariance) between source and target latent features
- spectral decoupling: introduce l2 penalty on logits $\hat y_i$

## Training
- Bert: 6 transformer blocks, 512 hidden units and 8 attention heads and 64 batch size. 4 instances are sampled per label when number of labels are small and 1 sample per label when it is large. 
- label-wise attention network on top of bert encoder, which uses one attention head per label to generate document representation. 
- lr is 2e-5, 64 batch sizes and 20 epochs using early stopping on the valid set. run 3 experiments with different random seed for each configuration. 

## Evaluation
- mean R-Precision: 
- micro-F1 and macro-F1: assess class-wise performance disparity

### Review Process

- [ ] I have already reviewed this paper
